dhis-2/dhis-api/src/main/java/org/hisp/dhis/scheduling/SchedulingManager.java

```diff
@@ -113,6 +113,13 @@
      * @param job The job to be executed
      */
     void executeJob( Runnable job );
+    
+    /**
+     * Schedule an ad hoc job without validation
+     * @param job The job to be scheduled
+     * @param startTime The time at which the job should start
+     */
+    void scheduleJob( Runnable job , Date startTime);
```

Comment:
Is it possible to make this take in a JobConfiguration object? If it has inMemoryJob toggled there is no validation, but it runs through DefaultJobInstance which will make things more consistent.

It would also be nice if you rename it to scheduleJobAtTime(or something)

Reply:
It is more like an ad-hoc job. Similar to 
    /**
     * Execute an actual job without validation
     *
     * @param job The job to be executed
     */
    void executeJob( Runnable job );

I am not sure why we dont have a JobConfiguration parameter in the above method? I can add jobConfiguration to scheduleJob(). I also felt "scheduleJob(.., Date date) is kind of self explanatory. Maybe we can change it to scheduleJobWithStartTime()